src/ios/CDVLocation.m

```diff
@@ -94,61 +94,62 @@ - (BOOL)isLocationServicesEnabled
 
 - (void)startLocation:(BOOL)enableHighAccuracy
 {
-    if (![self isLocationServicesEnabled]) {
-        [self returnLocationError:PERMISSIONDENIED withMessage:@"Location services are not enabled."];
-        return;
-    }
-    if (![self isAuthorized]) {
-        NSString* message = nil;
-        BOOL authStatusAvailable = [CLLocationManager respondsToSelector:@selector(authorizationStatus)]; // iOS 4.2+
-        if (authStatusAvailable) {
-            NSUInteger code = [CLLocationManager authorizationStatus];
-            if (code == kCLAuthorizationStatusNotDetermined) {
-                // could return POSITION_UNAVAILABLE but need to coordinate with other platforms
-                message = @"User undecided on application's use of location services.";
-            } else if (code == kCLAuthorizationStatusRestricted) {
-                message = @"Application's use of location services is restricted.";
+    [self.commandDelegate runInBackground:^{
+        if (![self isLocationServicesEnabled]) {
```

Comment:
I think we are abusing of the weak references to self in the plugins. In most cases it's not really needed because we don't retain the block http://stackoverflow.com/questions/20030873/always-pass-weak-reference-of-self-into-block-in-arc